packages/firestore/src/local/indexeddb_index_manager.ts

```diff
@@ -1062,3 +1031,23 @@ function indexStateStore(
 ): SimpleDbStore<DbIndexStateKey, DbIndexState> {
   return getStore<DbIndexStateKey, DbIndexState>(txn, DbIndexStateStore);
 }
+
+function getMinOffsetFromFieldIndexes(fieldIndexes: FieldIndex[]): IndexOffset {
+  hardAssert(
+    fieldIndexes.length !== 0,
+    'Found empty index group when looking for least recent index offset.'
+  );
+
+  let minOffset: IndexOffset = fieldIndexes[0].indexState.offset;
+  let maxBatchId: number = minOffset.largestBatchId;
+  for (let i = 1; i < fieldIndexes.length; i++) {
```

Comment:
I'm curious why you switched from `for (const fieldIndex of fieldIndexes) {` ?
I _think_ that's the preferred way in JS if you don't really care about the loop index value

Reply:
but now you're assuming the array has at least 2 elements, right? `fieldIndexes[1]` shouldn't be accessed if we are trying to find the minOffset of an array of size 1.